src/main/java/seedu/duke/commands/ExchangeCommand.java

```diff
@@ -123,4 +124,28 @@ public BigDecimal parseAmount () throws InvalidNumberException {
             throw new InvalidNumberException();
         }
     }
+
+    /**
+    * Prints a message indicating if any required accounts are missing.
+    *
+    * @param ui the user interface to use for printing the message
+    * @param accounts the list of accounts to check for missing accounts
+    */
+    private void printMissingAccounts (Ui ui, AccountList accounts) {
+        try {
+            Forex exchangeRate = formatInput();
+            try {
+                accounts.getAccount(exchangeRate.getInitial());
+            } catch (NoAccountException f) {
+                ui.printMessage("You need a " + exchangeRate.getInitial() + " account!");
+            }
+            try {
+                accounts.getAccount(exchangeRate.getTarget());
```

Comment:
While this code works to find out an account that does not exist, exceptions should only be used to designate a negative path for the intention of the method. The getAccount method is not being used for its intended purpose, instead its exception is used as the success path in a sense.
I suggest instead of using exceptions and this method to find the missing account, you could create a boolean method in the AccountList that determines the existence of an account.